packages/core/src/__tests__/ceramic-api.test.ts

```diff
@@ -238,7 +238,7 @@ describe('Ceramic API', () => {
         })
       }
 
-      expect(logRecords).toEqual(expected)
+      expect(JSON.stringify(logRecords)).toEqual(JSON.stringify(expected))
```

Comment:
Why is this change needed?

Reply:
The test kept failing on my machine and IntelliJ identified that as missing JSONification since the serialized objects were essentially the same. I'll take this change out to see if it works and if it was just Windows being Windows.

Reply:
Could be because of cached records but this is what I'm seeing on my machine (even in WSL):
```
 FAIL  src/__tests__/ceramic-api.test.ts (76.816 s)
  ● Ceramic API › API › can list log records

    expect(received).toEqual(expected) // deep equality

    Compared values serialize to the same structure.
    Printing internal object structure without calling `toJSON` instead.

    - Expected  - 0
    + Received  + 1

    @@ -41,10 +41,11 @@
                  62,
                  114,
                  84,
                ],
                "version": 1,
    +           Symbol(@ipld/js-cid/CID): true,
              },
              "payload": "AXESICWDwB7sbj3IWGh3QnQLkoFxfjPhWa-m-AQc34wPPnJU",
              "signatures": Array [
                Object {
                  "protected": "eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa2pLZUg4U2dWQVlDdlRCb3l4eDd1UkpGR00yYTlIVWVGd2ZKZmQ2Y3R1QTNYI3o2TWtqS2VIOFNnVkFZQ3ZUQm95eHg3dVJKRkdNMmE5SFVlRndmSmZkNmN0dUEzWCJ9",

      239 |       }
      240 |
    > 241 |       expect(logRecords).toEqual(expected)
          |                          ^
      242 |     })
      243 |
      244 |     it('can store commit if the size is lesser than the maximum size ~256KB', async () => {

      at Object.<anonymous> (src/__tests__/ceramic-api.test.ts:241:26)
```

Reply:
Yep, seeing the same failure in CircleCI after reverting this change and also switching to the existing LRU cache library.

Reply:
We've seen issues like this before. I think it might have to do with mismatched versions of the CID library.  @ukstv may be able to provide some insight and guidance here

Reply:
does this test fail on your machine against a clean build of the `develop` branch?

Reply:
No, it works otherwise and only fails with the cache. You might have hit it on the nose with the CID equality comment @stbrody. Let me either switch to strings or look at a custom comparator and try again.

Reply:
Ok, I figured it out. I changed code in `loadStreamCommits` in `ceramic.ts` like so:

```
-         const record = (await this.ipfs.dag.get(cid, { timeout: IPFS_GET_TIMEOUT })).value
+         const record = await this.dispatcher.retrieveCommit(cid)
```

I did this because the dispatcher is available and there's no reason not to use the cache. Once I updated the test code to also use `Dispatcher.retrieveCommit()` instead if `IpfsApi.dag` to obtain `expected`.

In other words, it wasn't the introduction of the cache causing this issue but a mismatched comparison.